### ISSUE TITLE
Bug: HomeLeave animation triggers on a .gpx with missing data.

### ISSUE THREAD

**Issue by mateuszrymar** [mateuszrymar/gpx-analyzer#45]
**Bug report**
To reproduce:
1. Upload a .gpx without time data.

🐛 What's the unexpected behaviour currently?
After loading an invalid file, homeLeave animation fires. So all tiles and a warning disappear, then a loading bar appears. It loads to 100% and stays there forever. The user can wait for a long time until he/she will notice there's something wrong.

👍 What's the expected behaviour?
The error should stay on screen and the loading bar should not show up at all.

🔧 Possible fixes:
Look into when and where homeLeave animation is run. Make it run **after** gpx validation.

**Comment by mateuszrymar** [comment]
🔧 Suggested solution is not very good.

**TLDR: we need to prioritize common use cases, not an edge case.**

The animation is fired **before** the .gpx is read, so we can do only minor checks on the file - we can check the size, extension but not the content.
This helps UX a lot, because user gets immediate feedback from the app that the app is doing **something**. If we run the animation after the check, user on a low-end mobile sees a frozen screen for 5s or more - so he'll quit the app.


Here's an alternative approach:
Make another animation that will interrupt the running loading screen and will bring back all home elements.
